pcdsdevices/sample_delivery.py

```diff
@@ -0,0 +1,435 @@
+    SP1 = UCpt(EpicsSignal)
+    set_SP1 = UCpt(EpicsSignal)
+    current1 = UCpt(EpicsSignal)
+
+    temperature2 = UCpt(EpicsSignal)
+    SP2 = UCpt(EpicsSignal)
+    set_SP2 = UCpt(EpicsSignal)
+    current2 = UCpt(EpicsSignal)
+
+    reboot = UCpt(EpicsSignal)
+
+    def __init__(self, prefix, *, name, **kwargs):
+        UCpt.collect_prefixes(self, kwargs)
+        super().__init__(prefix, name=name, **kwargs)
+
+
+class HPLC(Device):
+    """
+    An HPLC for the sample delivery system.
+
+    Parameters
+    ----------
+    prefix : str
+        The base PV for the Pressure Controller.
+
+    name : str
+        A name for the device.
+
+    status_prefix : str
+        Status of the HPLC.
+
+    run_prefix : str
+        Run the HPLC.
+
+    flowrate_prefix : str
+        Flow rate of the HPLC.
+
+    set_flowrate_prefix : str
+        Set the flow rate of the HPLC.
+
+    flowrate_SP_prefix : str
+        Set point for the flow rate.
+
+    pressure_prefix : str
+        Pressure in the HPLC.
+
+    pressure_units_prefix : str
+        Units for the pressure.
+
+    set_max_pressure_prefix : str
+        Set the maximum pressure.
+
+    max_pressure_prefix : str
+        Maximum pressure.
+
+    clear_error_prefix : str
+        Clear errors.
+    """
+
+    status = UCpt(EpicsSignal)
+    run = UCpt(EpicsSignal)
+
+    flowrate = UCpt(EpicsSignal)
+    set_flowrate = UCpt(EpicsSignal)
+    flowrate_SP = UCpt(EpicsSignal)
+
+    pressure = UCpt(EpicsSignal)
+    pressure_units = UCpt(EpicsSignal)
+    set_max_pressure = UCpt(EpicsSignal)
+    max_pressure = UCpt(EpicsSignal)
+
+    clear_error = UCpt(EpicsSignal)
+
+    def __init__(self, prefix, *, name, **kwargs):
+        UCpt.collect_prefixes(self, kwargs)
+        super().__init__(name=name, **kwargs)
+
+
+class PressureController(Device):
+    """
+    A Pressure Controller for the sample delivery system.
+
+    Parameters
+    ----------
+    prefix : str
+        The base PV for the Pressure Controller.
+
+    name : str
+        A name for the device.
+    """
+
+    status = Cpt(PytmcSignal, ':IO:SyncUnitOK', io='i', kind='normal')
+
+    pressure1 = Cpt(PytmcSignal, ':PropAir1:Pressure', io='i', kind='normal')
+    enabled1 = Cpt(PytmcSignal, ':PropAir1:Enable', io='io', kind='normal')
+    SP1 = Cpt(PytmcSignal, ':PropAir1:Setpoint', io='io', kind='normal')
+    low_limit1 = Cpt(PytmcSignal, ':PropAir1:LowLimit', io='io', kind='normal')
+    high_limit1 = Cpt(PytmcSignal, ':PropAir1:HighLimit', io='io',
+                      kind='normal')
+
+    pressure2 = Cpt(PytmcSignal, ':PropAir2:Pressure', io='i', kind='normal')
+    enabled2 = Cpt(PytmcSignal, ':PropAir2:Enable', io='io', kind='normal')
+    SP2 = Cpt(PytmcSignal, ':PropAir2:Setpoint', io='io', kind='normal')
+    low_limit2 = Cpt(PytmcSignal, ':PropAir2:LowLimit', io='io', kind='normal')
+    high_limit2 = Cpt(PytmcSignal, ':PropAir2:HighLimit', io='io',
+                      kind='normal')
```

Comment:
Worth making a proportionair subdevices?  Strictly speaking these have more signals that the `ViciValve` device :P

Reply:
Haha, I guess so